app/cdash/include/sendemail.php

```diff
@@ -134,26 +128,28 @@ function check_email_update_errors($buildid)
 }
 
 /** Return a summary for a category of error */
-function get_email_summary($buildid, $errors, $errorkey, $maxitems, $maxchars, $testtimemaxstatus, $emailtesttimingchanged)
+function get_email_summary(int $buildid, array $errors, $errorkey, int $maxitems, int $maxchars, int $testtimemaxstatus, bool $emailtesttimingchanged): string
 {
-    $config = Config::getInstance();
     $build = new Build();
     $build->Id = $buildid;
 
     $serverURI = get_server_URI();
     $information = '';
 
     // Update information
-    if ($errorkey == 'update_errors') {
+    if ($errorkey === 'update_errors') {
         $information = "\n\n*Update*\n";
 
         $buildUpdate = new BuildUpdate();
         $buildUpdate->BuildId = $buildid;
         $update = $buildUpdate->GetUpdateForBuild(PDO::FETCH_OBJ);
+        if ($update === false) {
+            throw new RuntimeException('Error querying update status for build ' . $buildid);
+        }
 
-        $information .= "Status: {$update->status} ({$serverURI}/viewUpdate.php?buildid={$buildid})\n";
```

Comment:
mostly just curious, but why is this an array now when we asked for an object? (`PDO::FETCH_OBJ`)

Reply:
This is one of the first files I did, and I probably didn't realize what `PDO::FETCH_OBJ` did.  I'll revert the changes since they're unnecessary.